libs/designsystem/src/lib/components/modal/services/modal.controller.ts

```diff
@@ -80,12 +80,18 @@ export class ModalController implements OnDestroy {
     await this.showAndRegisterOverlay(() => this.modalHelper.showModalWindow(config), onClose);
   }
 
-  public async navigateToModal(path: string | string[]): Promise<boolean> {
-    return this.modalNavigationService.navigateToModal(path);
+  public async navigateToModal(
+    path: string | string[],
+    queryParams?: Params | null
+  ): Promise<boolean> {
+    return this.modalNavigationService.navigateToModal(path, queryParams);
   }
 
-  public async navigateWithinModal(relativePath: string): Promise<boolean> {
-    return this.modalNavigationService.navigateWithinModal(relativePath);
+  public async navigateWithinModal(
+    relativePath: string,
+    queryParams?: Params | null
```

Comment:
Why the null type? 🤔
```suggestion
    queryParams?: Params
```